tests/x11/gnomecase/login_test.pm

```diff
@@ -30,7 +30,12 @@ sub run {
     assert_screen "generic-desktop";
     $self->auto_login_alter;
     my $ov = get_var('NOAUTOLOGIN');
-    set_var('NOAUTOLOGIN', '');
+    if (!get_var('NOAUTOLOGIN')) {
```

Comment:
One could simply use negation ov $ov here, as what you are trying to do is to flip value of NOAUTOLOGIN.
```perl 
set_var('NOAUTOLOGIN', !$ov)
```